Scripts/Player_Scripts/player.gd

```diff
@@ -13,13 +19,45 @@ func _input(event):
 		
 func _ready():
 	Input.set_mouse_mode(Input.MOUSE_MODE_CAPTURED)
-	
 
 func _physics_process(delta: float) -> void:
 	_apply_gravity(delta)
 	_jump()
 	_handle_movement()
 	move_and_slide()
+	
+	# Sticking flashlight code here for now
+	# Stops charging down once charge is at 0
+	if (light_charge > 0 && flash_light_on == true):
+		light_charge -= delta
+	
+	# Checks if light charge is under the fading threshold, default 40%
+	# then starts fading the flashlight
+	if (flash_light_on == true):
+		if (light_charge < light_fade_threshold):
+			if(light_charge < 0):
+				light_charge = 0
+			flash_light.light_energy = (light_charge / light_fade_threshold)
+		else:
+			flash_light.light_energy = 1
+	else:
+		flash_light.light_energy = 0
+	
+	if (Input.is_action_just_pressed("right_click")):
+		_flash_light_toggle()
+
+func _flash_light_toggle():
+	if (flash_light_on == true):
+		flash_light_on = false
+	else:
+		flash_light_on = true
```

Comment:
No Changes for this, just wanted to let you know for future that you could just do something like:

func _flash_light_toggle():
    flash_light_on = !flash_light_on

that will have the exact same functionallity as your function.